### PR TITLE
nav: init at 1.2.1

### PR DESCRIPTION
**Pull request by David-Kopczynski**
Hello :)

I added `nav` from https://github.com/Jojo4GH/nav with the functionality of navigating files in the terminal more easily.

While [Jojo4GH](https://github.com/Jojo4GH) is the project maintainer, I will help him keep this package up to date on nix. This is the reason, I added both of us to the maintainers list.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/na/nav/package.nix
Comment (steeleduncan) [inline pkgs/by-name/na/nav/package.nix:45]: with lib is deprecated #208242

File: pkgs/by-name/na/nav/package.nix
Comment (steeleduncan) [inline pkgs/by-name/na/nav/package.nix:58]: `platforms = platforms.linux` would be simpler here

File: pkgs/by-name/na/nav/package.nix
Comment (steeleduncan) [inline pkgs/by-name/na/nav/package.nix:26]: Could this not be built from source, using, e.g. `gradle2nix` rather than downloading the release binary?

**Review by steeleduncan — CHANGES_REQUESTED** [review]

File: pkgs/by-name/na/nav/package.nix
Comment (David-Kopczynski) [reply pkgs/by-name/na/nav/package.nix:26]: While looking for `gradle2nix`, I could not find any documentation in this repository at first and thus tried to get it running with [gradle](https://nixos.org/manual/nixpkgs/unstable/#gradle). However, we could not get the build running (our latest config is shown below).

This is the reason, I looked into `gradle2nix` again. The repository is currently migrating to [V2](https://github.com/tadfisher/gradle2nix/tree/v2) with functionality that is not (yet?) available for nixpkgs. We are currently wondering how to achieve the build and would like to know whether to stick to the binaries or if we should use:

```nix
let
  gradle2nix = import (fetchTarball "https://github.com/tadfisher/gradle2nix/archive/v2.tar.gz")  {};
in
gradle2nix.buildGradlePackage { ... }
```
*(as described in the documentation for `gradle2nix`)*

Would love to know how to proceed or where to get further information.

<details>
<summary>Nix Config for gradle.fetchDeps</summary>

We tried many variations of this file with `gradleUpdateTask`, `gradleUpdateScript`, and `gradleFlags`, but all of them failed at the dependency download or building step. @Jojo4GH's theory was that the dependency resolution does not work for `common` architectures, failing at downloading `kotlin-stdlib:2.0.0` for `arm64`.

```nix
{
  stdenv,
  lib,
  fetchFromGitHub,
  gradle,
  makeWrapper,
  jre,
  nix-update-script,
}:

stdenv.mkDerivation (finalAttrs: rec {
  pname = "nav";
  version = "1.2.1";

  src = fetchFromGitHub {
    owner = "Jojo4GH";
    repo = "nav";
    rev = "v${version}";
    hash = "sha256-rS64lvS8lBpOsNq+abqqLrX5HJsPTdVx6Akjg9S/pyQ=";
  };

  nativeBuildInputs = [
    gradle
  ];

  mitmCache = gradle.fetchDeps {
    inherit (finalAttrs) pname;
    data = ./deps.json;
  };

  __darwinAllowLocalNetworking = true;

  installPhase = ''
    runHook preInstall

    # this was never reached...

    runHook postInstall
  '';

  passthru.updateScript = nix-update-script { };

  meta = {
    description = "The Interactive and Stylish Replacement for ls & cd";
    longDescription = ''
      To make use of nav, add the following lines to your configuration:
      `programs.bash.shellInit = "eval \"$(nav --init bash)\"";` and
      `programs.zsh.shellInit = "eval \"$(nav --init zsh)\"";`
    '';
    homepage = "https://github.com/Jojo4GH/nav";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [
      David-Kopczynski
      Jojo4GH
    ];
    platforms = lib.platforms.linux;
    sourceProvenance = with lib.sourceTypes; [
      fromSource
      binaryBytecode # for gradle dependencies
    ];
    mainProgram = "nav";
  };
})
```
</details>

**Review by steeleduncan — APPROVED** [review]

File: pkgs/by-name/na/nav/package.nix
Comment (steeleduncan) [reply pkgs/by-name/na/nav/package.nix:26]: I've left an approving review as I don't see any real problem with packaging it as you have, and I wouldn't want to do anything to discourage someone with the commit bit merging it.

I'm curious though, so I'll have a try packaging with the `v2` gradle2nix, and if successful that can go as a separate PR once this is merge

File: pkgs/by-name/na/nav/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/na/nav/package.nix:46]:
```suggestion
    description = "Interactive and stylish replacement for ls & cd";
```

File: pkgs/by-name/na/nav/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/na/nav/package.nix:30]:
```suggestion
    stdenv.cc.cc.lib
```

File: pkgs/by-name/na/nav/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/na/nav/package.nix:11]: You can use `stdenv.hostPlatform.parsed.cpu.name` instead

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by David-Kopczynski** [comment]
Thanks @FliegendeWurst for the improvements :)

**Review by FliegendeWurst — APPROVED** [review]

**Comment by David-Kopczynski** [comment]
@steeleduncan I will look into `gradle2nix` once I see further activity. Apart from that, ofborg seems to be finally happy with my commit :)

**Pull request merged**
